database/kv/etcd/v3/client.go

```diff
@@ -211,35 +234,98 @@ func (c *Client) GetEndPoints() []string {
 	return c.endpoints
 }
 
-// if k not exist will put k/v in etcd, otherwise return nil
-func (c *Client) put(k string, v string, opts ...clientv3.OpOption) error {
+// if k not exist will put k/v in etcd, otherwise return ErrCompareFail
+func (c *Client) create(k string, v string, opts ...clientv3.OpOption) error {
 	rawClient := c.GetRawClient()
 
 	if rawClient == nil {
 		return ErrNilETCDV3Client
 	}
 
-	_, err := rawClient.Txn(c.ctx).
-		If(clientv3.Compare(clientv3.Version(k), "<", 1)).
+	resp, err := rawClient.Txn(c.ctx).
+		If(clientv3.Compare(clientv3.CreateRevision(k), "=", 0)).
 		Then(clientv3.OpPut(k, v, opts...)).
 		Commit()
+
+	if err != nil {
+		return err
+	}
+	if !resp.Succeeded {
+		return ErrCompareFail
+	}
+	return nil
+}
+
+// if k in bulk insertion not exist all, then put all k/v in etcd, otherwise return error
+func (c *Client) batchCreate(kList []string, vList []string, opts ...clientv3.OpOption) error {
+	rawClient := c.GetRawClient()
+
+	if rawClient == nil {
+		return ErrNilETCDV3Client
+	}
+
+	kLen := len(kList)
+	vLen := len(vList)
+	if kLen == 0 || vLen == 0 || kLen != vLen {
+		return ErrKVListSizeIllegal
+	}
+
+	var cs []clientv3.Cmp
+	var ops []clientv3.Op
+
+	for i, k := range kList {
+		v := vList[i]
+		cs = append(cs, clientv3.Compare(clientv3.CreateRevision(k), "=", 0))
+		ops = append(ops, clientv3.OpPut(k, v, opts...))
+	}
+
+	resp, err := rawClient.Txn(c.ctx).
+		If(cs...).
+		Then(ops...).
```

Comment:
may be put these a line, seems better

Reply:
@georgehao I do not think it is a problem.